autoload/pythonimports.vim

```diff
@@ -6,12 +6,35 @@ function! pythonimports#filename2module(filename)
   " directly included in PYTHONPATH.
   let pkg = fnamemodify(a:filename, ":p")
   let root = fnamemodify(pkg, ":h")
-  while strlen(root)
-    if !filereadable(root . "/__init__.py")
-      break
-    endif
-    let root = fnamemodify(root, ":h")
-  endwhile
+
+  let found = 0
+  if exists("g:python_paths")
+      for path in g:python_paths
+            if root =~ path
+                "path detected
+                while strlen(root)
+                    if root == path
+                        "echo 'found '. root
+                        let found = 1
+                        break
+                    endif
+                    let root = fnamemodify(root, ":h")
+                endwhile
+            endif
+            if found
+                break
+            endif
```

Comment:
This constant checking of `if found | break` is a bit tedious, how about extracting the loop into a helper function that can do an early `return`?

Reply:
(This is a suggestion, not a requirement for merge.)